core/shared/src/main/scala/scalaz/zio/internal/StackBool.scala

```diff
@@ -27,30 +27,32 @@ private[zio] final class StackBool private () {
   private[this] var _size = 0L
 
   final def getOrElse(index: Int, b: Boolean): Boolean = {
-    var j   = index.toLong
+    val i0  = _size & 63L
+    val i   = ((64L - i0) + index) & 63L
+    var ie  = ((64L - i0) + index) >> 6
     var cur = head
-    while (j >= 256L && (cur.next ne null)) {
-      j -= 256L
+
+    while (ie > 0) {
+      ie -= 1
       cur = cur.next
```

Comment:
```
val stack = StackBool()
stack.getOrElse(300, false)
```
NullPointerException

Reply:
Thanks for spotting this. I've added a null check here.